client/src/pages/onboarding/steps/SetupConnection.tsx

```diff
@@ -51,22 +51,18 @@ export const SetupConnection: React.FC<Props> = ({
     }
   }, [connectionId])
 
-  useInterval(
-    () => {
-      if (outOfBandId && document.visibilityState === 'visible') {
-        dispatch(fetchConnectionByOutOfBandId(outOfBandId))
+  useWebhookEvent(
+    ConnectionEventTypes.ConnectionStateChanged,
+    (event: { payload: { connectionRecord: ConnectionRecord } }) => {
+      if (
+        event.payload.connectionRecord.outOfBandId === outOfBandId ||
+        event.payload.connectionRecord.id === connectionId
+      ) {
+        dispatch(setConnection(event.payload.connectionRecord))
       }
     },
-    !connectionId ? 1000 : null
-  )
-
-  useInterval(
-    () => {
-      if (connectionId && document.visibilityState === 'visible') {
-        dispatch(fetchConnectionById(connectionId))
-      }
-    },
-    !isCompleted && connectionId ? 1000 : null
+    !connectionId || (!isCompleted && connectionId ? true : false),
```

Comment:
I'm not sure what does does exactly: `(!isCompleted && connectionId ? true : false)`. Can you help me understand?

Reply:
Checks if isCompleted is false and connectionId exists, returns true if condition evaluates to true and vice versa. It is check to ensure that the hook runs if the connectionId exists but for some reason, isCompleted is still false. You know we're listening for events, so we need to make sure that the hook triggers even when the connection has been been received,  but the offer has not been accepted